content/automate/ManageIQ/Transformation/StateMachines/VMTransformation.class/__methods__/update_status.rb

```diff
@@ -0,0 +1,119 @@
+module Transformation
+  module StateMachines
+    module VMTransformation
+      class UpdateStatus
+        def initialize(handle = $evm)
+          @handle = handle
+        end
+
+        # This method gets the current state, even in the context of nested
+        # state machines. Where running a nested state machines, there can be
+        # more than one active states: one per level of nesting. The real
+        # active state is the one with the deepest ancestry.
+        def current_state(states)
```

Comment:
@fdupont-redhat 
Couldn't you get the current state from root['ae_state']?
We are setting it here https://github.com/ManageIQ/manageiq-automation_engine/blob/master/lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_state_machine.rb#L63

Reply:
In the context of nested state machines, `root [ae_state]` contains the name of the current state in the current state machine. And nothing prevents from having two states with the same name when using nested state machines. So, here I take the state ancestry depth as a criteria to distinguish the real running state.

Reply:
@fdupont-redhat 
So do you need the fully qualified instance name that is being used at that moment? We have that information we can set it as an extra attribute in the root object.